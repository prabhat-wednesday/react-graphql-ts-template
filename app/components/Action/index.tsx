import React from 'react';

interface InputSearchBoxProps {
  onChange: React.ChangeEventHandler<HTMLInputElement>;
}

export const InputSearchBox = ({ onChange }: InputSearchBoxProps) => {
  return (
    <div>
      <input onChange={onChange} type="text" />
    </div>
  );
};
