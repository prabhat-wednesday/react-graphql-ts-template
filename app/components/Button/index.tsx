import { ButtonProps, Button as But } from 'antd';
import React from 'react';
import styled from 'styled-components';
import { T } from '@components/T';

const B = styled(But)`
  margin: 0.5em;
`;
interface CustomButtonProps extends ButtonProps {
  messageId: string;
}
export const Button = (props: CustomButtonProps) => {
  return (
    <B {...props}>
      <T id={props.messageId} />
    </B>
  );
};
