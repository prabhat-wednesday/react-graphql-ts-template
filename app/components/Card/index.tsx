import React from 'react';
import { Card, Row } from 'antd';
import styled from 'styled-components';
import { Song } from '@app/containers/ItuensContainer/types';
const { Meta } = Card;

export interface LoadAbleCardProps {
  dataToShow: Song[];
}

const CustomRow = styled(Row)`
  background-color: antiquewhite;
  gap: 2rem;

  @media (max-width: 320px) {
    justify-content: center;
  }
`;

const H1 = styled.h1`
  font-size: 1rem;
`;

const CustomCard = styled(Card)`
  padding: 5px;
  margin: 1rem;
  width: 17rem;

  @media (max-width: 320px) {
    width: 12rem;
    margin: 0.5rem;
  }
`;

const LoadAbleCard = ({ dataToShow }: LoadAbleCardProps) => {
  return (
    <CustomRow>
      {dataToShow.map(({ trackId, artistName, artworkUrl100, collectionName }) => (
        <CustomCard hoverable key={trackId} cover={<img src={artworkUrl100} loading="lazy" />}>
          <H1>Artist Name: {artistName}</H1>
          <Meta title={collectionName} />
        </CustomCard>
      ))}
    </CustomRow>
  );
};

export default LoadAbleCard;
