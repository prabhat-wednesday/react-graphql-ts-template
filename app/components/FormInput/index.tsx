import React from 'react';
import { Form, Input } from 'antd';
import styled from 'styled-components';
import { translate } from '../IntlGlobalProvider';

const StyledInput = styled(Input)`
  width: 100%;
`;

const CustomFormItem = styled(Form.Item)`
  display: flex;
  justify-content: space-between;

  &&&.ant-form-horizontal .ant-form-item-label {
    margin-right: 10px;
  }
`;
interface FormInputProps {
  label: string;
  name: string;
  message: string;
  required: boolean;
  onChange?: (e: React.ChangeEvent<HTMLInputElement>) => void;
}
const FormInput = ({ label, name, message, required, onChange }: FormInputProps) => {
  return (
    <CustomFormItem label={translate(label)} name={translate(name)} rules={[{ required, message: translate(message) }]}>
      <StyledInput onChange={onChange} data-testid="formInput" />
    </CustomFormItem>
  );
};

export default FormInput;
