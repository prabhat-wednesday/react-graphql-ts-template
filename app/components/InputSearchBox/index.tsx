import React from 'react';
import { Input } from 'antd';
import styled from 'styled-components';

interface InputSearchBoxProps {
  onChange: React.ChangeEventHandler<HTMLInputElement>;
  searchLabel: string;
}

const InputContainer = styled.div`
  display: flex;
  flex-direction: column;
  align-items: center;
  margin: 0.5rem auto;
`;

const CustomLabel = styled.label`
  display: block;
  font-size: 1.5rem;
`;

const CustomInput = styled(Input)`
  width: 30%;
  display: block;

  @media (max-width: 320px) {
    width: 80%;
  }
`;

export const InputSearchBox = ({ onChange, searchLabel }: InputSearchBoxProps) => {
  if (searchLabel === '') {
    searchLabel = 'Search here';
  }
  return (
    <InputContainer>
      <CustomLabel>{searchLabel}</CustomLabel>
      <CustomInput onChange={onChange} type="text" placeholder="search here" />
    </InputContainer>
  );
};
