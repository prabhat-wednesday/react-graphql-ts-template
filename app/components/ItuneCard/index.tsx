import React, { useRef, useEffect } from 'react';
import { Card } from 'antd';
import styled from 'styled-components';
import { media } from '@app/themes';
import If from '../If';

const { Meta } = Card;

export interface ItuneCardProps {
  trackId: number;
  artistName: string;
  artworkUrl100: string;
  collectionName: string;
  previewUrl: string;
  currentTrackId: number;
  handleOnPlay: (number: any) => any;
}

const CustomCard = styled(Card)`
  && {
    padding: 5px;
    margin: 1rem;
    width: 17rem;

    ${media.lessThan('tablet')`
      width: 12rem;
      margin: 0.5rem;
    `}
  }
`;

const StyledImg = styled.img`
  && {
    max-height: 15rem;
    object-fit: cover;
  }
`;

const StyledAudio = styled.audio`
  && {
    width: 100%;
    padding: 5px;
  }
`;

const StyledMeta = styled(Meta)`
  && {
    padding: 5px;
  }
`;

const ItuneCard = ({
  trackId,
  artistName,
  artworkUrl100,
  collectionName,
  previewUrl,
  currentTrackId,
  handleOnPlay
}: ItuneCardProps) => {
  const audioRef = useRef(null);
  useEffect(() => {
    if (audioRef.current) {
      if (currentTrackId !== trackId) {
        audioRef.current?.pause();
      }
    }
  }, [currentTrackId]);
  return (
    <If condition={trackId}>
      <CustomCard
        hoverable
        key={trackId}
        bodyStyle={{ padding: '10px' }}
        cover={<StyledImg src={artworkUrl100} loading="lazy" data-testid="cover-img" />}
      >
        <StyledMeta title={artistName} description={collectionName} />
        <StyledAudio
          data-testid="audio-element"
          onPlay={() => handleOnPlay(trackId)}
          ref={audioRef}
          controls
          src={previewUrl}
        >
          <a href={previewUrl}>Audio Preview</a>
        </StyledAudio>
      </CustomCard>
    </If>
  );
};

export default React.memo(ItuneCard);
