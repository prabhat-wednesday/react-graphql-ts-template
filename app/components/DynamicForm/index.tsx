import React, { useEffect } from 'react';
import { Form as F, FormInstance } from 'antd';
import styled from 'styled-components';
import FormInput from '@components/FormInput';
import { StepperLoginFormInputInterface } from '@app/interfaces/common';

const FormContainer = styled.div`
  display: flex;
  flex: 1 1 0;
  align-self: center;
  padding: 1rem;
`;

const Form = styled(F)`
  width: 20rem;
`;
interface DynamicStepFormProps {
  formInput: StepperLoginFormInputInterface;
  handleFormInstance: (value: FormInstance<any>) => void;
  handleSubmit: (value: any) => void;
}

const DynamicStepForm = ({ formInput, handleSubmit, handleFormInstance }: DynamicStepFormProps) => {
  const [form] = Form.useForm();

  useEffect(() => {
    handleFormInstance(form);
  }, [form]);

  const onFinish = (values: any) => {
    handleSubmit(values);
    form.resetFields();
  };

  return (
    <FormContainer>
      <Form
        id="my-form"
        layout="vertical"
        data-testid="login-form"
        form={form}
        name="basic"
        initialValues={{ remember: true }}
        onFinish={onFinish}
        autoComplete="off"
      >
        <FormInput {...formInput} />
      </Form>
    </FormContainer>
  );
};

export default DynamicStepForm;
