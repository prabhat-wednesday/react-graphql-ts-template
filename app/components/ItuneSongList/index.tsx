import React from 'react';
import { Row } from 'antd';
import { get } from 'lodash-es';
import styled from 'styled-components';
import { Song } from '@app/containers/ItunesContainer/types';
import { colors, media } from '@app/themes';
import { For } from '../For';
import ItuneCard from '../ItuneCard';

interface ItuneSongListProps {
  songData: {
    results?: Song[];
  };
}

const CustomRow = styled(Row)`
  && {
    background-color: ${colors.cardContainerBg};
    gap: 2.5rem;

    ${media.lessThan('tablet')`
    justify-content: center;`}
  }
`;

const ItuneSongList = ({ songData }: ItuneSongListProps) => {
  const results = get(songData, 'results', []);
  return (
    <>
      <For ParentComponent={CustomRow} renderItem={(song: Song) => <ItuneCard {...song} />} of={results} />
    </>
  );
};

export default ItuneSongList;
