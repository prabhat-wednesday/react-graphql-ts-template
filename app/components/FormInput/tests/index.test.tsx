import React from 'react';
import { renderProvider, renderWithIntl, timeout } from '@app/utils/testUtils';
import FormInput from '..';
import { fireEvent } from '@testing-library/react';
import { translate } from '@app/components/IntlGlobalProvider';

describe('<FormInput/> component test', () => {
  const onChangeSpy = jest.fn();
  const FormInputProps = {
    label: 'Username',
    name: 'username',
    message: 'Please input your username!',
    onChange: onChangeSpy,
    required: true
  };
  it('should render and match the snapshot', () => {
    const { baseElement } = renderWithIntl(<FormInput {...FormInputProps} />);
    expect(baseElement).toMatchSnapshot();
  });

  it('should match the label name present in the FormInput ', () => {
    const { getByLabelText } = renderWithIntl(<FormInput {...FormInputProps} />);
    expect(getByLabelText(translate('Username'))).toBeInTheDocument();
  });

  it('should check if form input have data when onchange is fire', async () => {
    const { getByTestId } = renderProvider(<FormInput {...FormInputProps} />);
    const LoginInput = getByTestId('formInput');
    fireEvent.change(LoginInput, {
      target: { value: 'Prabhat' }
    });
    await timeout(500);
    expect(onChangeSpy).toBeCalled();
  });
});
