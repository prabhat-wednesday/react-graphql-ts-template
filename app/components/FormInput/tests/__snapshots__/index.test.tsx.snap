// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FormInput/> component test should render and match the snapshot 1`] = `
<body>
  <div>
    <div
      class="ant-row ant-form-item FormInput__CustomFormItem-ofxmhy-1 cElTSv"
    >
      <div
        class="ant-col ant-form-item-label"
      >
        <label
          class="ant-form-item-required"
          for="username"
          title="Username"
        >
          Username
        </label>
      </div>
      <div
        class="ant-col ant-form-item-control"
      >
        <div
          class="ant-form-item-control-input"
        >
          <div
            class="ant-form-item-control-input-content"
          >
            <input
              class="ant-input FormInput__StyledInput-ofxmhy-0 iWohJH"
              data-testid="formInput"
              id="username"
              type="text"
              value=""
            />
          </div>
        </div>
      </div>
    </div>
  </div>
</body>
`;
